src/psyclone/psyir/nodes/operation.py

```diff
@@ -117,9 +127,12 @@ def is_elemental(self):
         return self.operator not in self._non_elemental_ops
 
     def __str__(self):
-        result = self.node_str(False) + "\n"
-        for entity in self._children:
-            result += str(entity) + "\n"
+        result = f"{self.node_str(False)}\n"
+        for idx, entity in enumerate(self._children):
+            if self._named_args and self._named_args[idx]:
```

Comment:
There are some similar expressions (e.g. in the fortran backend and for Calls as well as operations) where we get the index from enumerating the children and use it in the named args, I put the discussion here but it applies in all the other cases as well:

1) As mentioned in a previous comment I think the `if self._named_args` can be removed if we initialise to an empty list

2) The self._named_args[idx] may not exist. The  `if self._named_args`  offered some protection but I think it is insufficient as it assumes the named args have been set always at the end. But since PSyIR is mutable we could have added a new children after setting the named args and now the two list contain a different number of items.

There are 2 solutions, or we put the necessary IndexError exceptions and/or checks the both lens are the same everywhere that it is used or we guarantee inside the class the named_args is always the same length than children.

I would favor the later. Since we access it through a property, we could extend the property to cut or extend the _named_args list with None to always match the number of children. This would make the use of name_args much safer.

Then we need to replace internal accesses to use the property as well (like here) to have the same guarantees.

I still have the doubt that if we detach() a specific node or insert in a specific location, that aren't the end of the list. Should the given name args be tied to were they are, or shifted? It is a decision to make but tied to were they are is the easiest to implement.

Reply:
I've implemented your prefered solution (with your input). I've made it so that named args are shifted if args are added or removed as this makes most sense to me.

Reply:
I also needed to add support for copy() as the id's stored in _named_args may be inconsistent on entry to the copy (so need to be made consistent) and will be inconsistent on exit from the copy (so again need to be made consistent).